internal/storage/blob.go

```diff
@@ -0,0 +1,8 @@
+package storage
+
+// TODO(adam): There might be a problem with implementing m.isolateMergableDir()
```

Comment:
`gsutil` can do it but it looks as though they're doing it by [renaming each object individually](https://cloud.google.com/storage/docs/gsutil/commands/mv#renaming-groups-of-objects) (so non-atomically as you pointed out):

> You can use the gsutil mv command to rename all objects with a given prefix to have a new prefix.
...
> Unlike the case with many file systems, the gsutil mv command does not perform a single atomic operation. Rather, it performs a copy from source to destination followed by removing the source for each object.

It says as much a bit farther down the page.

Reply:
Yea. I need to investigate this more. The subtle differences would cause issues in production. We still need this PR for https://github.com/moov-io/achgateway/issues/53 so I'm planning to merge it still.